report_thesis/src/sections/methodology/testing_validation.tex

```diff
@@ -0,0 +1,223 @@
+
+The fold assignments are then merged back into the original dataset (Line~\ref{line:merge_folds}), and the dataset is split into a test set (consisting of the highest fold number) and the remaining training data (Line~\ref{line:split_dataset}).
+The training data is further divided into \( k-1 \) folds for cross-validation.
+For each fold, the training data consists of all but one fold, which serves as the validation set.
+These pairs of training and validation sets are then appended to the list of cross-validation folds \(\mathbf{F}_\text{cv}\) (Line~\ref{line:create_folds}).
+
+Finally, the fold indicator column is removed from all datasets before returning the final partitions (Line~\ref{line:remove_fold_column}).
+This cleanup step ensures that the fold information does not interfere with subsequent data processing or model training.
+
+The final output of this procedure consists of:
+\begin{itemize}
+    \item The cross-validation folds \(\mathbf{F}_\text{cv}\), each containing a tuple of training and validation sets.
+    \item The training set \(\mathbf{D}_\text{train}\) in its entirety.
+    \item The test set \(\mathbf{D}_\text{test}\), distinct from the training set.
+\end{itemize}
+
+By following this detailed procedure, we ensure that our cross-validation is robust against data leakage, maintains the integrity of grouped targets, and carefully handles extreme values to improve the representativeness of the training set.
+
+Our method is broadly aligned with the approach described by \citet{andersonImprovedAccuracyQuantitative2017}.
+They employed a similar strategy to assess the performance of their PLS model, using k-fold cross-validation and a separate test set.
+Their process involved dividing the full set of laboratory data into five folds, using four for cross-validation and combining them as the final training set, while the fifth fold served as a test set.
+This approach ensured that each fold represented the full elemental compositional variation, and extreme targets were included in the training folds to handle a wider range of compositions.
+For consistency, we also use \(k=5\) for our cross-validation.
+
+Additionally, by using $k=5$ folds, we have effectively chosen an 80\%/20\% split between the training and testing datasets.
+In our experience, this ratio maximizes the training set's capacity for effective model learning while ensuring that the testing set is sufficiently representative to provide an accurate assessment of the model's performance on new data.
+As detailed in Section~\ref{sec:problem_definition}, the limited availability of data is a primary constraint.
+Allocating too much data to the testing set could compromise the comprehensiveness of the training set, undermining the model's ability to generalize effectively.
```

Comment:
That was actually being mentioned in the Evaluation Metrics section. We can add the following line to further clarify the implications.

But just to clarify, we actually get higher RMSE_CV values, as the 'hard to predict values' are placed in the training set, which the folds based on. The RMSEP (RMSE on test set) is actually lower, since it doesn't deviate much from the mean.

```diff
However, our data partitioning method, which moves the most extreme values into the training data, naturally results in the testing data being closer to the mean of the data distribution, making it easier to predict.
+ In practice, this would result in lower \texttt{rmsep} and \texttt{std\_dev} values compared to the cross validation metrics.
Therefore, evaluating the model's performance using both the cross-validation metrics (\texttt{rmse\_cv} and \texttt{std\_dev\_cv}) and the test set metrics (\texttt{rmsep} and \texttt{std\_dev}) is crucial.
The cross-validation metrics provide insights into the model's stability across different subsets, while the test set metrics offer a final measure of performance on truly unseen data, giving a comprehensive assessment of the model's generalizability.
```

Reply:
Added